src/game/board/boardSituation.js

```diff
@@ -1,14 +1,31 @@
-export default class BoardSituation {
-    constructor() {
+export default class BoardSituation {
+        constructor(firstBoardPosition) {
+            if(newBoardPosition instanceof  BoardPosition){
+                this.boardPosition=firstBoardPosition;
+                this.boardSituation=[];
```

Comment:
Nazwa "boardSituation" jest myląca. Proponuję "_previousSituation" (mamy do tego getter, to typowa konwencja, żeby prywatna wartość z dostępem poprzez getter/setter nazywała się tak samo, tylko z myślnikiem).